### ISSUE TITLE
SQL Field Filter parameter substitution does unnecessary casting for temporal columns

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#19751]
In #14835, #15115, and elsewhere we optimized the way we generate filter clauses against temporal columns to avoid unnecessary casts  (amongst other changes) to better hit column indexes, e.g.

```sql
WHERE date BETWEEN ? AND ?
```

as opposed to

```sql
WHERE cast(date AS date) BETWEEN ? AND ?
```

However these changes did not cover SQL Field Filters. We should tweak the SQL parameters code to get the performance optimizations there as well

**Comment by camsaul** [comment]
This will be fixed in #19384

**Comment by flamber** [comment]
Looks like a duplicate of #17711

**Comment by camsaul** [comment]
Fixed by #19384

**Comment by camsaul** [comment]
Duplicate of #17711
